### PR TITLE
Add message preview with attachments in channel list

### PR DESCRIPTION
**Pull request by nuno-vieira**
### 🔗 Issue Links
https://github.com/GetStream/ios-issues-tracking/issues/550

### 🎯 Goal
Add message attachment previews in the Channel List. This is a feature that is already present in the other SDKs. 

### 🎨 Showcase
The snapshots are a good way to see how message previews are rendered with attachments.

### 🧪 Manual Testing Notes
1. Open DemoApp
2. Open a Channel
3. Send a message with an attachment
4. Go to the Channel List
5. The attachment should be visible in the message preview

**Note**: It should be tested for Image, Video, File and Giphy attachment.

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should be manually QAed
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

File: Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView.swift [hunk 1]
```diff
@@ -368,4 +372,32 @@ extension ChatChannelListItemView {
...
+    /// - Parameter previewMessage: The preview message of the channel.
+    /// - Returns: A string representing the message preview text.
+    func attachmentPreviewText(for previewMessage: ChatMessage) -> String? {
+        guard let attachment = previewMessage.allAttachments.first else {
+            return nil
+        }
+        let text = previewMessage.textContent ?? previewMessage.text
+        switch attachment.type {
+        case .audio:
+            return "🎧 \(text.isEmpty ? "Audio" : text)"
```
Comment (martinmitrevski) [inline Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView.swift:386]: All the texts here should be localized, not hardcoded to English.

File: Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView.swift [hunk 2]
```diff
@@ -368,4 +372,32 @@ extension ChatChannelListItemView {
     var isLastMessageVoiceRecording: Bool {
         content?.channel.previewMessage?.voiceRecordingAttachments.isEmpty == false && typingUserString == nil
     }
+
+    /// The message preview text in case it contains attachments.
+    /// - Parameter previewMessage: The preview message of the channel.
+    /// - Returns: A string representing the message preview text.
+    func attachmentPreviewText(for previewMessage: ChatMessage) -> String? {
```
Comment (martinmitrevski) [inline Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView.swift:379]: Does it make sense for this to be customizable by customers?

**Review by martinmitrevski — COMMENTED** [review]
lgtm, just added few comments

File: Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView.swift (hunk 1 above)
Comment (nuno-vieira) [reply Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView.swift:386]: Good catch 👍

File: Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView.swift (hunk 2 above)
Comment (nuno-vieira) [reply Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView.swift:379]: Yeah, I think it kind of does. They can do it anyway if they override the subtitle text, and it should be easy as well. For that reason, I didn't open it up yet. My plan was to keep it internal for now, and if a customer really needs it, we can make it open. What do you think?

File: Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView.swift (hunk 1 above)
Comment (nuno-vieira) [reply Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView.swift]: Solved ✅

**Review by martinmitrevski — APPROVED** [review]
LGTM!

**Pull request merged**